Android Gradle build error on modern Java

### Search before asking

- [x] I have searched https://github.com/ultralytics/yolo-flutter-app/issues and did not find a similar report.


### Project area

Android build

### Bug

The Android plugin build forces Java version 17 here:

https://github.com/ultralytics/yolo-flutter-app/blob/bfce3422f9440aa972350320a8bd179d1e2cf6cf/android/build.gradle#L46-L60

This means that for most devs that have a more modern Java toolchain (21 or 25) the Android build with the Ultralytics YOLO Flutter package fails:

```text
Cannot find a Java installation on your machine matching:
{languageVersion=17, vendor=any vendor, implementation=vendor-specific, nativeImageCapable=false}
```

I presume this is a bug to hard-require ``JavaLanguageVersion.of(17)` since the old toolchain requirement is not mentioned in your docs.

Can you please relax the toolchain requirement so that the package works on modern dev machines?



### Environment

I have this:
```
 flutter doctor
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.41.4, on macOS 26.3.1 25D2128 darwin-arm64, locale en-US)
[✓] Android toolchain - develop for Android devices (Android SDK version 35.0.1)
[✓] Xcode - develop for iOS and macOS (Xcode 26.3)
```

and get this error
```
FAILURE: Build failed with an exception.

* What went wrong:
Could not determine the dependencies of task ':ultralytics_yolo:bundleLibCompileToJarDebug'.
> Could not create task ':ultralytics_yolo:compileDebugJavaWithJavac'.
   > Failed to calculate the value of task ':ultralytics_yolo:compileDebugJavaWithJavac' property 'javaCompiler'.
      > Cannot find a Java installation on your machine (Mac OS X 26.3.1 aarch64) matching: {languageVersion=17, vendor=any vendor, implementation=vendor-specific, nativeImageCapable=false}. Toolchain download repositories have not been configured.
```

### Minimal Reproducible Example

Follow your docs to create started project.

### Additional

_No response_

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,5 +1,7 @@
 ## 0.2.0
 
+- **Feature**: YOLO26 model release — `yolo26*` CoreML (`.mlpackage`) and TFLite (`.tflite`) artifacts published alongside YOLOv11 (#431, #432).
+- **Enhancement**: YOLO property and method surface updates (#428).
 - Documentation update for model downloads:
   - All model download links point to GitHub release `v0.2.0` (no bundled models).
   - Added YOLO26 references alongside YOLOv11 for both CoreML (`.mlpackage`) and TFLite (`.tflite`) artifacts.

--- a/README.md
+++ b/README.md
@@ -12,9 +12,6 @@
 
 Welcome to the Ultralytics YOLO Flutter plugin! Integrate cutting-edge [Ultralytics YOLO](https://docs.ultralytics.com/) [computer vision](https://www.ultralytics.com/glossary/computer-vision-cv) models seamlessly into your Flutter mobile applications. This plugin at https://pub.dev/packages/ultralytics_yolo supports both Android and iOS platforms, offering APIs for [object detection](https://docs.ultralytics.com/tasks/detect/), [image classification](https://docs.ultralytics.com/tasks/classify/), [instance segmentation](https://docs.ultralytics.com/tasks/segment/), [pose estimation](https://docs.ultralytics.com/tasks/pose/), and [oriented bounding box detection](https://docs.ultralytics.com/tasks/obb/).
 
-<!-- ![YOLO Flutter Demo](https://via.placeholder.com/600x300/1e1e1e/ffffff?text=YOLO+Flutter+Demo+GIF+Coming+Soon) -->
-<!-- TODO: Add actual demo GIF/video -->
-
 **✨ Why Choose YOLO Flutter?**
 
 | Feature         | Android | iOS |
@@ -71,7 +68,7 @@ YOLOView(
 
 ```yaml
 dependencies:
-  ultralytics_yolo: ^0.1.26
+  ultralytics_yolo: ^0.2.0
 ```
 
 ### 2. Install dependencies
@@ -102,7 +99,7 @@ YOLO("yolo11n.pt").export(format="coreml", nms=True)
 YOLO("yolo11n-seg.pt").export(format="coreml")
 ```
 
-**[📥 Download Models](doc/install.md#models)**
+**[📥 Download Models](doc/models.md)**
 
 Bundle the model with your app using the following method.
 

--- a/analysis_options.yaml
+++ b/analysis_options.yaml
@@ -45,7 +45,6 @@ analyzer:
     missing_return: warning
     todo: ignore
     avoid_print: warning
-    public_member_api_docs: true
   exclude:
     - "**/*.g.dart"
     - "**/*.freezed.dart"

--- a/android/build.gradle
+++ b/android/build.gradle
@@ -52,13 +52,6 @@ android {
         jvmTarget = JavaVersion.VERSION_17.toString() // Use toString() for jvmTarget
     }
 
-    // Configure Java Toolchain to use JDK 17
-    java {
-        toolchain {
-            languageVersion = JavaLanguageVersion.of(17)
-        }
-    }
-
     sourceSets {
         main.java.srcDirs += "src/main/kotlin"
         test.java.srcDirs += "src/test/kotlin"

--- a/doc/api.md
+++ b/doc/api.md
@@ -19,26 +19,34 @@ class YOLO {
   YOLO({
     required String modelPath,
     required YOLOTask task,
+    bool useGpu = true,
     bool useMultiInstance = false,
+    Map<String, dynamic>? classifierOptions,
+    int? numItemsThreshold,
   });
 }
 ```
 
 #### Constructor Parameters
 
-| Parameter          | Type       | Required | Default | Description                           |
-| ------------------ | ---------- | -------- | ------- | ------------------------------------- |
-| `modelPath`        | `String`   | ✅       | -       | Path to the YOLO model file (.tflite) |
-| `task`             | `YOLOTask` | ✅       | -       | Type of YOLO task to perform          |
-| `useMultiInstance` | `bool`     | ❌       | `false` | Enable multi-instance support         |
+| Parameter           | Type                    | Required | Default | Description                                           |
+| ------------------- | ----------------------- | -------- | ------- | ----------------------------------------------------- |
+| `modelPath`         | `String`                | ✅       | -       | Path to the YOLO model file                           |
+| `task`              | `YOLOTask`              | ✅       | -       | Type of YOLO task to perform                          |
+| `useGpu`            | `bool`                  | ❌       | `true`  | Enable GPU acceleration when supported                |
+| `useMultiInstance`  | `bool`                  | ❌       | `false` | Enable multi-instance support                         |
+| `classifierOptions` | `Map<String, dynamic>?` | ❌       | `null`  | Optional classifier preprocessing and label overrides |
+| `numItemsThreshold` | `int?`                  | ❌       | `30`    | Maximum number of returned detections                 |
 
 #### Properties
 
-| Property     | Type       | Description                              |
-| ------------ | ---------- | ---------------------------------------- |
-| `instanceId` | `String`   | Unique identifier for this YOLO instance |
-| `modelPath`  | `String`   | Path to the loaded model file            |
-| `task`       | `YOLOTask` | Current task type                        |
+| Property        | Type       | Description                              |
+| --------------- | ---------- | ---------------------------------------- |
+| `instanceId`    | `String`   | Unique identifier for this YOLO instance |
+| `isInitialized` | `bool`     | Whether the model has been loaded        |
+| `modelPath`     | `String`   | Path to the loaded model file            |
+| `task`          | `YOLOTask` | Current task type                        |
+| `useGpu`        | `bool`     | Whether GPU acceleration is enabled      |
 
 #### Methods
 
@@ -162,6 +170,20 @@ static Future<Map<String, String?>> getStoragePaths()
 
 **Returns**: Map of storage location names to paths
 
+###### `withClassifierOptions()`
+
+Create a YOLO instance configured for classification models that need custom preprocessing.
+
+```dart
+static YOLO withClassifierOptions({
+  required String modelPath,
+  required YOLOTask task,
+  required Map<String, dynamic> classifierOptions,
+  bool useGpu = true,
+  bool useMultiInstance = false,
+})
+```
+
 ---
 
 ### YOLOTask Enum
@@ -198,31 +220,43 @@ class YOLOView extends StatefulWidget {
     required this.modelPath,
     required this.task,
     this.controller,
+    this.cameraResolution = "720p",
     this.onResult,
     this.onPerformanceMetrics,
     this.onStreamingData,
+    this.showNativeUI = false,
     this.onZoomChanged,
-    this.cameraResolution = "720p",
-    this.showNativeUI = true,
     this.streamingConfig,
+    this.confidenceThreshold = 0.5,
+    this.iouThreshold = 0.45,
+    this.useGpu = true,
+    this.showOverlays = true,
+    this.overlayTheme = const YOLOOverlayTheme(),
+    this.lensFacing = LensFacing.back,
   }) : super(key: key);
 }
 ```
 
 #### Constructor Parameters
 
-| Parameter              | Type                                | Required | Default  | Description                                             |
-| ---------------------- | ----------------------------------- | -------- | -------- | ------------------------------------------------------- |
-| `modelPath`            | `String`                            | ✅       | -        | Path to YOLO model file (camera starts even if invalid) |
-| `task`                 | `YOLOTask`                          | ✅       | -        | YOLO task type                                          |
-| `controller`           | `YOLOViewController?`               | ❌       | `null`   | Custom view controller                                  |
-| `onResult`             | `Function(List<YOLOResult>)?`       | ❌       | `null`   | Detection results callback                              |
-| `onPerformanceMetrics` | `Function(YOLOPerformanceMetrics)?` | ❌       | `null`   | Performance metrics callback                            |
-| `onStreamingData`      | `Function(Map<String, dynamic>)?`   | ❌       | `null`   | Comprehensive streaming callback                        |
-| `onZoomChanged`        | `Function(double)?`                 | ❌       | `null`   | Zoom level change callback                              |
-| `cameraResolution`     | `String`                            | ❌       | `"720p"` | Camera resolution                                       |
-| `showNativeUI`         | `bool`                              | ❌       | `true`   | Show native camera UI                                   |
-| `streamingConfig`      | `YOLOStreamingConfig?`              | ❌       | `null`   | Streaming configuration                                 |
+| Parameter              | Type                                | Required | Default                    | Description                                             |
+| ---------------------- | ----------------------------------- | -------- | -------------------------- | ------------------------------------------------------- |
+| `modelPath`            | `String`                            | ✅       | -                          | Path to YOLO model file (camera starts even if invalid) |
+| `task`                 | `YOLOTask`                          | ✅       | -                          | YOLO task type                                          |
+| `controller`           | `YOLOViewController?`               | ❌       | `null`                     | Custom view controller                                  |
+| `cameraResolution`     | `String`                            | ❌       | `"720p"`                   | Camera resolution                                       |
+| `onResult`             | `Function(List<YOLOResult>)?`       | ❌       | `null`                     | Detection results callback                              |
+| `onPerformanceMetrics` | `Function(YOLOPerformanceMetrics)?` | ❌       | `null`                     | Performance metrics callback                            |
+| `onStreamingData`      | `Function(Map<String, dynamic>)?`   | ❌       | `null`                     | Comprehensive streaming callback                        |
+| `showNativeUI`         | `bool`                              | ❌       | `false`                    | Show native camera controls                             |
+| `onZoomChanged`        | `Function(double)?`                 | ❌       | `null`                     | Zoom level change callback                              |
+| `streamingConfig`      | `YOLOStreamingConfig?`              | ❌       | `null`                     | Streaming configuration                                 |
+| `confidenceThreshold`  | `double`                            | ❌       | `0.5`                      | Initial confidence threshold for YOLOView               |
+| `iouThreshold`         | `double`                            | ❌       | `0.45`                     | Initial IoU threshold for YOLOView                      |
+| `useGpu`               | `bool`                              | ❌       | `true`                     | Enable GPU acceleration for camera inference            |
+| `showOverlays`         | `bool`                              | ❌       | `true`                     | Draw Flutter-side detection overlays                    |
+| `overlayTheme`         | `YOLOOverlayTheme`                  | ❌       | `const YOLOOverlayTheme()` | Customize Flutter overlay styling                       |
+| `lensFacing`           | `LensFacing`                        | ❌       | `LensFacing.back`          | Initial camera lens selection                           |
 
 #### Example
 
@@ -239,7 +273,7 @@ YOLOView(
   },
 )
 
-// Camera-only mode (v0.1.25+): starts even with invalid model path
+// Camera-only mode: starts even with invalid model path
 YOLOView(
   modelPath: 'model_not_yet_downloaded.tflite',  // Model doesn't exist yet
   task: YOLOTask.detect,
@@ -344,7 +378,7 @@ Parameters:
 
 - `PlatformException` - If model file cannot be found or loaded
 
-**Note**: As of v0.1.25, YOLOView can start with an invalid model path (camera-only mode). Use this method to load a valid model later.
+**Note**: YOLOView can start with an invalid model path (camera-only mode). Use this method to load a valid model later.
 
 Example:
 
@@ -364,6 +398,62 @@ try {
 } catch (e) {
   print('Failed to load model: $e');
 }
+```
+
+##### `zoomIn()`
+
+Increase camera zoom by one step.
+
+```dart
+Future<void> zoomIn()
+```
+
+##### `zoomOut()`
+
+Decrease camera zoom by one step.
+
+```dart
+Future<void> zoomOut()
+```
+
+##### `setZoomLevel()`
+
+Set the camera zoom directly.
+
+```dart
+Future<void> setZoomLevel(double zoomLevel)
+```
+
+##### `stop()`
+
+Stop the active camera session.
+
+```dart
+Future<void> stop()
+```
+
+##### `restartCamera()`
+
+Restart the camera session after stopping it.
+
+```dart
+Future<void> restartCamera()
+```
+
+##### `setShowUIControls()`
+
+Show or hide the native camera controls.
+
+```dart
+Future<void> setShowUIControls(bool show)
+```
+
+##### `setShowOverlays()`
+
+Show or hide Flutter-side detection overlays.
+
+```dart
+Future<void> setShowOverlays(bool show)
 ```
 
 ##### `setStreamingConfig()`

--- a/doc/install.md
+++ b/doc/install.md
@@ -16,7 +16,7 @@ Add the plugin to your Flutter project's `pubspec.yaml`:
 dependencies:
   flutter:
     sdk: flutter
-  ultralytics_yolo: ^0.1.18 # Latest version
+  ultralytics_yolo: ^0.2.0 # Latest version
 ```
 
 Run the installation command:
@@ -67,11 +67,11 @@ Edit `android/app/build.gradle`:
 ```gradle
 // android/app/build.gradle
 android {
-    compileSdkVersion 34
+    compileSdkVersion 35
 
     defaultConfig {
-        minSdkVersion 24  // Minimum API level 24 required
-        targetSdkVersion 34
+        minSdkVersion 21  // Minimum API level 21 required
+        targetSdkVersion 35
     }
 }
 ```
@@ -194,9 +194,9 @@ android {
 | Platform    | Minimum Version | Recommended   |
 | ----------- | --------------- | ------------- |
 | **iOS**     | 13.0+           | 14.0+         |
-| **Android** | API 24+         | API 28+       |
-| **Flutter** | 3.3.0+          | Latest stable |
-| **Dart**    | 3.0.0+          | Latest stable |
+| **Android** | API 21+         | API 28+       |
+| **Flutter** | 3.32.1+         | Latest stable |
+| **Dart**    | 3.8.1+          | Latest stable |
 
 ## 🔍 Troubleshooting Installation
 

--- a/doc/quickstart.md
+++ b/doc/quickstart.md
@@ -33,8 +33,8 @@ Add to `pubspec.yaml`:
 dependencies:
   flutter:
     sdk: flutter
-  ultralytics_yolo: ^0.1.25
-  image_picker: ^0.8.7 # For image selection
+  ultralytics_yolo: ^0.2.0
+  image_picker: ^1.2.1 # For image selection
 ```
 
 Install dependencies:
@@ -53,7 +53,7 @@ You can get the model in one of the following ways:
 
 3. Export it from [Ultralytics/ultralytics](https://github.com/ultralytics/ultralytics) ([CoreML](https://docs.ultralytics.com/ja/integrations/coreml/)/[TFLite](https://docs.ultralytics.com/integrations/tflite/))
 
-**[📥 Download Models](./install.md#models)** |
+**[📥 Download Models](./models.md)** |
 
 Bundle the model with your app using the following method.
 

--- a/doc/troubleshooting.md
+++ b/doc/troubleshooting.md
@@ -69,16 +69,16 @@ end
 
 #### Minimum SDK Version Error
 
-**Symptoms**: `uses-sdk:minSdkVersion 16 cannot be smaller than version 24`
+**Symptoms**: `uses-sdk:minSdkVersion 16 cannot be smaller than version 21`
 
 **Solution**: Update `android/app/build.gradle`:
 
 ```gradle
 android {
     defaultConfig {
-        minSdkVersion 24  // Change from 16 to 24
-        targetSdkVersion 34
-        compileSdkVersion 34
+        minSdkVersion 21  // Change from 16 to 21
+        targetSdkVersion 35
+        compileSdkVersion 35
     }
 }
 ```
@@ -291,7 +291,7 @@ await controller.switchModel(modelPath, YOLOTask.detect);
 - Model loading failed silently
 - Invalid model format
 
-**Solution**: As of version 0.1.25, the plugin supports camera-only mode. If model loading fails, camera will continue without inference. Check logs for model loading errors:
+**Solution**: The plugin supports camera-only mode. If model loading fails, camera will continue without inference. Check logs for model loading errors:
 
 ```
 iOS: "YOLOView Warning: Model file not found"
@@ -654,7 +654,7 @@ class DebugProfiler {
 Future<Map<String, dynamic>> collectDebugInfo() async {
   return {
     'flutter_version': 'Run: flutter --version',
-    'plugin_version': '0.1.25',
+    'plugin_version': '0.2.0',
     'platform': Platform.isIOS ? 'iOS' : 'Android',
     'model_path': 'yolo11n',
     'model_exists': await YOLO.checkModelExists('yolo11n'),
@@ -667,7 +667,7 @@ Future<Map<String, dynamic>> collectDebugInfo() async {
 ### Community Support
 
 - **Discord**: [Ultralytics Discord](https://discord.com/invite/ultralytics)
-- **GitHub Issues**: [Flutter Plugin Issues](https://github.com/ultralytics/ultralytics/issues)
+- **GitHub Issues**: [Flutter Plugin Issues](https://github.com/ultralytics/yolo-flutter-app/issues)
 - **Documentation**: [Official Docs](https://docs.ultralytics.com)
 
 ## 🎯 Quick Fixes Checklist

--- a/example/pubspec.yaml
+++ b/example/pubspec.yaml
@@ -9,25 +9,23 @@ environment:
   flutter: ">=3.32.1"
 
 dependencies:
+  archive: ^4.0.9
+  cupertino_icons: ^1.0.9
   flutter:
     sdk: flutter
   flutter_launcher_icons: ^0.14.4
-  image_picker: ^0.8.7
-  path_provider: ^2.0.0
-  archive: ^3.0.0
-  http: ^1.0.0
-
+  http: ^1.6.0
+  image_picker: ^1.2.1
+  path_provider: ^2.1.5
   ultralytics_yolo:
     path: ../
 
-  cupertino_icons: ^1.0.8
-
 dev_dependencies:
-  integration_test:
-    sdk: flutter
+  flutter_lints: ^6.0.0
   flutter_test:
     sdk: flutter
-  flutter_lints: ^4.0.0
+  integration_test:
+    sdk: flutter
 
 flutter:
   uses-material-design: true

--- a/lib/core/yolo_inference.dart
+++ b/lib/core/yolo_inference.dart
@@ -185,7 +185,7 @@ class YOLOInference {
       // Reference: https://docs.ultralytics.com/reference/engine/results/
       final detection = <String, dynamic>{
         'classIndex': classificationMap['class'] ?? 0,
-        'className': classification['name'] ?? '',
+        'className': classificationMap['name'] ?? '',
         'confidence': MapConverter.safeGetDouble(
           classificationMap,
           'confidence',
@@ -210,7 +210,8 @@ class YOLOInference {
 
       for (final obb in obbList) {
         if (obb is Map) {
-          final points = obb['points'] as List<dynamic>? ?? [];
+          final obbMap = MapConverter.convertToTypedMap(obb);
+          final points = obbMap['points'] as List<dynamic>? ?? [];
 
           double minX = double.infinity, minY = double.infinity;
           double maxX = double.negativeInfinity, maxY = double.negativeInfinity;
@@ -229,11 +230,8 @@ class YOLOInference {
 
           final detection = <String, dynamic>{
             'classIndex': 0,
-            'className': obb['class'] ?? '',
-            'confidence': MapConverter.safeGetDouble(
-              MapConverter.convertToTypedMap(obb),
-              'confidence',
-            ),
+            'className': obbMap['class'] ?? '',
+            'confidence': MapConverter.safeGetDouble(obbMap, 'confidence'),
             'boundingBox': {
               'left': minX,
               'top': minY,

--- a/lib/utils/logger.dart
+++ b/lib/utils/logger.dart
@@ -4,6 +4,6 @@ import 'package:flutter/foundation.dart';
 
 void logInfo(String message) {
   if (kDebugMode) {
-    print('[YOLO DEBUG] $message');
+    debugPrint('[YOLO DEBUG] $message');
   }
 }

--- a/pubspec.yaml
+++ b/pubspec.yaml
@@ -25,14 +25,14 @@ environment:
 dependencies:
   flutter:
     sdk: flutter
-  plugin_platform_interface: ^2.0.2
+  plugin_platform_interface: ^2.1.8
 
 dev_dependencies:
-  coverage: ^1.13.1
+  coverage: ^1.15.0
   flutter_lints: ">=5.0.0 <7.0.0"
   flutter_test:
     sdk: flutter
-  mockito: ^5.4.4
+  mockito: ^5.6.4
   very_good_analysis: ">=6.0.0 <11.0.0"
 
 flutter:

--- a/test/yolo_model_manager_test.dart
+++ b/test/yolo_model_manager_test.dart
@@ -282,7 +282,6 @@ void main() {
 
       // Should not throw exception
       await manager.dispose();
-      expect(true, isTrue);
     });
 
     test('loadModel with default instance ID', () async {

--- a/test/yolo_test.dart
+++ b/test/yolo_test.dart
@@ -81,13 +81,6 @@ void main() {
       expect(platform, isNotNull);
     });
 
-    test('setModel calls method channel correctly', () async {
-      final platform = YOLOMethodChannel();
-      expect(platform, isNotNull);
-      // Test passes if no exceptions are thrown
-      expect(true, isTrue);
-    });
-
     test('platform interface works correctly', () {
       final mockPlatform = MockYOLOPlatform();
       expect(mockPlatform, isNotNull);
@@ -207,12 +200,6 @@ void main() {
       expect(segmentYolo.task, YOLOTask.segment);
       expect(classifyYolo.task, YOLOTask.classify);
       expect(poseYolo.task, YOLOTask.pose);
-    });
-  });
-
-  group('Error Handling', () {
-    test('handles platform exceptions gracefully', () async {
-      expect(true, isTrue);
     });
   });
 

--- a/test/yolo_view_test.dart
+++ b/test/yolo_view_test.dart
@@ -109,24 +109,6 @@ void main() {
       );
       expect(() => uninitializedController.switchCamera(), returnsNormally);
     });
-
-    test('error handling works correctly', () async {
-      controller.init(mockChannel, 1);
-
-      // Test that errors are handled gracefully
-      await controller.setConfidenceThreshold(0.8);
-      await controller.setIoUThreshold(0.6);
-      await controller.setNumItemsThreshold(50);
-      await controller.switchCamera();
-      await controller.zoomIn();
-      await controller.zoomOut();
-      await controller.setZoomLevel(2.0);
-      await controller.stop();
-      await controller.setShowUIControls(true);
-
-      // All methods should complete without throwing
-      expect(true, isTrue);
-    });
   });
 
   group('YOLOView Widget', () {
@@ -249,8 +231,7 @@ void main() {
         ),
       );
 
-      // Test passes if no exceptions are thrown
-      expect(true, isTrue);
+      expect(find.byType(YOLOView), findsOneWidget);
     });
 
     testWidgets('handles disposal correctly', (WidgetTester tester) async {
@@ -277,8 +258,7 @@ void main() {
       await tester.pumpAndSettle();
       await tester.pumpWidget(Container()); // Dispose the widget
 
-      // Test passes if no exceptions are thrown
-      expect(true, isTrue);
+      expect(find.byType(YOLOView), findsNothing);
     });
 
     testWidgets('fallback UI shown on unsupported platform', (

